src/main/java/edu/kit/compiler/transform/IRExpressionVisitor.java

```diff
@@ -0,0 +1,132 @@
+package edu.kit.compiler.transform;
+
+import edu.kit.compiler.data.AstVisitor;
+import edu.kit.compiler.data.DataType;
+import edu.kit.compiler.data.ast_nodes.ExpressionNode;
+import edu.kit.compiler.semantic.DefinitionKind;
+import firm.Construction;
+import firm.Mode;
+import firm.TargetValue;
+import firm.nodes.Node;
+
+import static edu.kit.compiler.data.ast_nodes.ExpressionNode.ValueExpressionType.IntegerLiteral;
+import static edu.kit.compiler.data.ast_nodes.ExpressionNode.ValueExpressionType.Null;
+
+/**
+ * return firm nodes for our ast nodes
+ */
+public class IRExpressionVisitor implements AstVisitor<Node> {
+
+    private final TransformContext context;
+
+    public IRExpressionVisitor(TransformContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.BinaryExpressionNode binaryExpressionNode) {
```

Comment:
You can explicitly import the nested classes. That would probably make your visitor a bit more concise.